src/sagas/search.js

```diff
@@ -0,0 +1,41 @@
+import { call, takeEvery, put, select } from 'redux-saga/effects';
+import search from '../actions/search';
+import { getSearch, getGif} from '../services/api'
+
+
+export function* loadSearchGifs(action) {
+  try {
+    const searchResponse = yield call(getSearch, action.payload);
+    yield put(search.creators.requestSearchGifsSuccess(searchResponse.data));
+  } catch (error) {
+    yield put(search.creators.requestSearchGifsFail(error));
+    console.log(error);
+  }
+}
+
+export function* loadSearchFavorites(action) {
```

Comment:
Nice!! You can use a function to select some slices of the state. Instead of using `yield select()` to get all the state object you could have done:

```
const selectFavs = state => state.favoriteGifs;
const favs = yield select(selectFavs);
```

Not sure about is the regex for though.

Reply:
I do the search with the regex, to achieve a `LIKE` functionality ala MySql